src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1834,8 +1834,13 @@ bool Creature::CreateFromProto(ObjectGuid::LowType guidlow, uint32 entry, Creatu
     }
 
     if (vehId)
+    {
         if (CreateVehicleKit(vehId, entry, true))
+        {
             UpdateDisplayPower();
+            m_movementInfo.pitch = m_vehicleKit->GetPitch();
```

Comment:
I would put this in `Vehicle::InitMovementInfoForBase`

Reply:
Updated